roles/kafka_broker/tasks/register_cluster.yml

```diff
@@ -32,8 +42,45 @@
       ]
     status_code: 204
   run_once: true
+  when: auth_mode != 'mtls'
   no_log: "{{mask_secrets|bool}}"
   register: output
   until: output.status == 204
   retries: "{{ mds_retries }}"
   delay: 10
+
+- name: Register Kafka Cluster mTLS
+  uri:
+    url: "{{mds_bootstrap_server_urls.split(',')[0]}}/security/1.0/registry/clusters"
+    method: POST
+    validate_certs: false
+    headers:
+      Content-Type: application/json
+      Authorization: "Bearer {{ authorization_token }}"
+    client_cert: "{{ kafka_broker_cert_path if send_client_cert|bool else omit }}"
+    client_key: "{{ kafka_broker_key_path if send_client_cert|bool else omit }}"
+    body_format: json
+    # Cant use internal listener as register cluster doesnt support SSL protocol
```

Comment:
could you share any reference/docs for this conclusion? if we are using mds endpoint, SSL should be sufficient afaik. SASL_SSL was needed for kafka rest api

Reply:
3rd point in this doc https://confluentinc.atlassian.net/wiki/spaces/OAAC/pages/3653567526/mTLS+with+RBAC+Gotchas

Reply:
Also to clarify the SASL_SSL protocol is needed in the body of this post request as the value of "protocol" parameter and doesnt mean the request to /registry/clusters endpoint must have some sasl credentials